langchain4j-core/src/main/java/dev/langchain4j/internal/GsonJsonCodec.java

```diff
@@ -59,6 +60,12 @@ public <T> T fromJson(String json, Class<T> type) {
         return GSON.fromJson(json, type);
     }
 
+    @Override
+    public <T, U> T fromJson(String json, Class<T> rawType, Class<U> typeArgumentClass) {
```

Comment:
```suggestion
    public <T> T fromJson(String json, Class<T> rawType, Class<?> typeArgumentClass) {
```
Seems like there is no need for `U`